src/main/java/org/broadinstitute/hellbender/tools/reference/ReferenceSequenceTable.java

```diff
@@ -0,0 +1,354 @@
+            columns.add(getReferenceDisplayName(path));
+        }
+        return columns;
+    }
+
+    public List<String> getColumnNames() {
+        return columnNames;
+    }
+
+    /**
+     * Given a GATKPath, return the name of the file as a String.
+     *
+     * @param reference The path to a reference.
+     * @return the name of the reference as a String.
+     */
+    public static String getReferenceDisplayName(GATKPath reference){
+        return reference.toPath().getFileName().toString();
+    }
+
+    public Map<String, Integer> getColumnIndices() {
+        return columnIndices;
+    }
+
+    /**
+     * Construct 2 tables of references: one keyed by MD5, one keyed by sequence name.
+     */
+    public void build() {
+        tableByMD5 = new LinkedHashMap<>();
+        tableBySequenceName = new LinkedHashMap<>();
+
+        for (Map.Entry<GATKPath, ReferenceDataSource> entry : referenceSources.entrySet()) {
+            SAMSequenceDictionary dictionary = entry.getValue().getSequenceDictionary();
+            for (SAMSequenceRecord record : dictionary.getSequences()) {
+                String name = record.getSequenceName();
+                int length = record.getSequenceLength();
+                String md5 = calculateMD5(record, entry.getValue());
+                GATKPath reference = entry.getKey();
+                TableEntry newEntry = new TableEntry(getReferenceDisplayName(reference), name);
+                TableRow newRow = null;
+
+                // map each MD5 to List of TableEntry objects containing length, md5, and name
+                if (!tableByMD5.containsKey(md5)) {
+                    newRow = new TableRow(md5, length, new ArrayList<>(referenceSources.keySet()));
+                    tableByMD5.put(md5, newRow);
+                }
+                tableByMD5.get(md5).add(newEntry);
+
+                if(!tableBySequenceName.containsKey(name)){
+                    Set<TableRow> rowSet = new HashSet<>();
+                    tableBySequenceName.put(name, rowSet);
+                }
+                tableBySequenceName.get(name).add(tableByMD5.get(md5));
+            }
+        }
+    }
+
+    public Set<String> getAllSequenceNames(){
+        return tableBySequenceName.keySet();
+    }
+
+    /**
+     * Given an MD5, returns its corresponding row
+     *
+     * @param md5 The MD5 as a String
+     * @return the corresponding TableRow from the tableByMD5
+     */
+    public TableRow queryByMD5(String md5){
+        return tableByMD5.get(md5);
+    }
+
+    /**
+     * Given a sequence name, returns the set of its corresponding rows
+     *
+     * @param sequenceName The sequence name as a String
+     * @return the set of TableRows that contain the sequence name
+     */
+    public Set<TableRow> queryBySequenceName(String sequenceName){
+        return tableBySequenceName.get(sequenceName) == null ? Collections.emptySet() : tableBySequenceName.get(sequenceName);
```

Comment:
Save the return value from `get()` in a local variable so that you don't have to call it twice.